vars/dockerExecute.groovy

```diff
@@ -229,7 +247,34 @@ void call(Map parameters = [:], body) {
     }
 }
 
+private waitForSidecarReadyOnDocker(String containerId, String command){
+    String dockerCommand = "docker exec ${containerId} ${command}"
+    waitForSidecarReady(dockerCommand)
+}
+
+private waitForSidecarReadyOnKubernetes(String containerName, String command){
+    container(name: containerName){
+        waitForSidecarReady(command)
+    }
+}
 
+private waitForSidecarReady(String command){
+    int sleepTimeInSeconds = 10
+    int timeoutInSeconds = 5 * 60
+    int maxRetries = timeoutInSeconds / sleepTimeInSeconds
+    int retries = 0
+    while(true){
+        String status = sh script:command, returnStatus:true
+        if(status == "0") return
+        if(retries > maxRetries){
+            error("Timeout while waiting for sidecar container to be ready")
```

Comment:
`timeout` step could be used.

Reply:
I am in favor of having control over the error message and as I anyway have to have a loop I don't see a big advantage.